src/Lecture1.hs

```diff
@@ -77,7 +85,18 @@ and 1 is the smallest, and 7 - 1 = 6.
 Try to use local variables (either let-in or where) to implement this
 function.
 -}
-minmax x y z = error "TODO"
+
+-- max :: Int -> Int -> Int
+-- max x y = if x < y then y else x
+
+-- min :: Int -> Int -> Int
+-- min x y = if x > y then y else x
+
+minmax :: Int -> Int -> Int -> Int
+minmax x y z = 
+    let maximum = max (max x y) z
```

Comment:
I see that GHC outputs some warnings ⚠️ 
In this case, it happens because `maximum` and `minimum` are the names of standard functions in Prelude. It still works due to name shadowing. But generally, it's better to avoid name conflicts with standard functions.

It's a bit tough to do this here because `max` and `maximum` are taken 😅 But I would use smth like `mx` or `maxVal` (and similar for minimum).

Reply:
Thank you very much, resolved naming problems for all such functions